Add options IGNORE_ARCHIVED_POSTS and IGNORE_REPLY_POSTS

seanthegeek opened the pull request:
Users importing old posts from Twitter/X can cause a feed to get flooded wold posts that match a keyword. To avoid this, ignore posts with a `created_at` timestamp that is older than 4 minutes.

MarshalX commented:
I am not sure that this is right approach. For example firehose could be down for 10 minutes. Then it will up again and start sending old posts. You need to process such posts and do not ignore. Another case is when your firehose consumer went down. And you want to backfill events. So you pass cursor back to 2 days ago. And start processing old events. These case a common

seanthegeek commented:
Maybe set it to a week?

MarshalX commented:
Looks like this is a specific feature. I do not think that it should be in the general template. Someone who needs it could implement by themself

seanthegeek commented:
How about this:

I removed the call to `is_archive_record` from the example filter, but kept the function in and added more documentation to it explaining the pros and cons. That should help new feed creators understand the issues.

That why, they understand what is going on when something like [this](https://bsky.app/profile/prevention-paradox.bsky.social/post/3lcpampq5k22s) happens

MarshalX commented:
I am okay to keep this function if we will provide an option to enable it (which is disabled by default); right now, it is dead code

MarshalX commented:
@seanthegeek thank you and sorry for the long review! happy new year!
